sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -113,6 +113,18 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
       }
     }
 
+    if (requiredSchema.length == 1 &&
+      requiredSchema.head.name == parsedOptions.columnNameOfCorruptRecord) {
+      throw new AnalysisException(
+        s"""
+           |'${parsedOptions.columnNameOfCorruptRecord}' cannot be selected alone without other
+           |data columns. If you want to select corrupt records only, cache or save the Dataset
+           |before executing queries. For example:
+           |df.cache() then
+           |df.select("${parsedOptions.columnNameOfCorruptRecord}")
+         """.stripMargin.replaceAll(System.lineSeparator(), " "))
```

Comment:
I wouldn't replace it but just concatenate the strings with `+`, or just keep the newlines. IIRC, I think using `System.lineSeparator()` wouldn't work on Windows anyway because the actual newlines of this string in the codes are `\n` but `System.lineSeparator()` gives `\r\n` up to me knowledge.

Reply:
agreed, will fix it in the next commit, thanks.